scipy/special/sf_error_state.cpp

```diff
@@ -1,10 +1,19 @@
 #include <stdlib.h>
-
 #include "sf_error_state.h"
 
+#ifdef __MINGW64__
+#include <mutex>
+std::mutex err_mutex;
+
+#define THREAD_LOCAL
+#else
+#define THREAD_LOCAL thread_local
+#endif
```

Comment:
This won't work. What thread-local define to use needs compiler checks - those are already implemented:

https://github.com/scipy/scipy/blob/7466219fb545a7b774c820ce22ae04826c59cd80/scipy/meson.build#L155-L194

You can include `scipy_config.h` and use `SCIPY_TLS` where you'd otherwise use `thread_local`.